app/scripts/services/tile-proxy.js

```diff
@@ -532,36 +527,54 @@ export const tileDataToPixData = (
 
 function text(url, callback) {
   /**
-   * Send a JSON request mark it so that we can tell how many are in flight
+   * Send a text request and mark it so that we can tell how many are in flight
    */
   requestsInFlight += 1;
   pubSub.publish('requestSent', url);
 
-  d3Text(url, (error, done) => {
-    callback(error, done);
-    pubSub.publish('requestReceived', url);
-    requestsInFlight -= 1;
-  });
+  fetch(url, {credentials: 'include', headers: headers})
+    .then(response => {
+      pubSub.publish('requestReceived', url);
+      if (response.ok) {
+        response.text().then(t => {
+          callback(null, t);
+          requestsInFlight -= 1;
+        })
+      } else {
+        // Do something?
+      }
+    })
+    .catch(error => {
```

Comment:
Flight counter is not updated. I'd suggest using `finally` to update it.

Reply:
I am done a super similar update here https://github.com/hms-dbmi/higlass/pull/307
Maybe we can merge the two efforts. :)

Reply:
How about if I make a branch off of your branch, and just duplicate what you've done for `text` for `json`? Thanks!